reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -450,22 +449,19 @@ private BuiltReportModule toBuiltReportModule(GAV gav, VersionLookupResult vlr)
                     CompletableFuture<Set<ProductArtifacts>> artifacts = productProvider.getArtifacts(ga);
                     artifacts = filterProductArtifacts(products, artifacts);
 
-                    //Needs to be re-worked
-                    artifacts.thenAccept((productArtifacts) -> gaProductArtifactsMap.put(ga, productArtifacts));
+                    // add the ga and its corresponding CompletableFuture for the artifacts in a map
+                    gaProductArtifactsMap.put(ga, artifacts);
         });
 
-        // Need to fill the map Map<GA, Set<ProductArtifacts>> gaProductArtifactsMap with data
-        // Should be executed in parallel
-
-
         /**  Use the mapping GA:Set<ProductArtifacts> and fill the LookupReports */
         request.getGavs().stream()
                 .distinct()
                 .forEach((gav) -> {
             LookupReport lr = new LookupReport(gav);
             reports.add(lr);
-            CompletableFuture<Set<ProductArtifacts>> artifacts = new CompletableFuture<>();
-            artifacts.supplyAsync(() -> gaProductArtifactsMap.get(gav.getGA())); //Need to check if it's correct
+
+            // get the CompletableFuture for the ga
+            CompletableFuture<Set<ProductArtifacts>> artifacts = gaProductArtifactsMap.get(gav.getGA());
 
```

Comment:
I think this approach, won't help us from reducing the number of calls to Indy. If I understand it correctly, for each GAV, we will do a request to Indy to get results of GA. I'm afraid, it's the same as before. To avoid the issues, we need to get the results for GA before, we do forEach on GAV.